manager/src/main/java/org/openremote/manager/security/KeyStoreServiceImpl.java

```diff
@@ -0,0 +1,194 @@
+			if(new File(defaultKeyStorePath.toUri()).exists()) {
+				this.keyStorePath = defaultKeyStorePath;
+				this.keyStore = KeyStore.getInstance(new File(defaultKeyStorePath.toUri()), getKeyStorePassword());
+			}else{
+				this.keyStore = createKeyStore(defaultKeyStorePath);
+			}
+
+			this.keyStorePath = defaultKeyStorePath;
+		}
+
+		if(trustStorePath.isPresent()){
+			this.keyStore = KeyStore.getInstance(trustStorePath.get().toFile(), trustStorePassword.toCharArray());
+		}else{
+			Path defaultTrustStorePath = persistenceService.resolvePath(Paths.get("keystores").resolve("client_truststore.p12"));
+			if(new File(defaultTrustStorePath.toUri()).exists()) {
+				this.trustStorePath = defaultTrustStorePath;
+				this.trustStore = KeyStore.getInstance(new File(defaultTrustStorePath.toUri()), getKeyStorePassword());
+			}else{
+				this.trustStore = createKeyStore(defaultTrustStorePath);
+			}
+			this.trustStorePath = defaultTrustStorePath;
+		}
+	}
+	private KeyStore getKeyStore() {
+		try {
+			return KeyStore.getInstance(this.keyStorePath.toAbsolutePath().toFile(), getKeyStorePassword());
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+	private KeyStore getTrustStore() {
+		try {
+			return KeyStore.getInstance(this.trustStorePath.toAbsolutePath().toFile(), getKeyStorePassword());
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private void storeKeyStore(KeyStore keystore) {
+		try {
+			this.keyStore = keystore;
+			keystore.store(new FileOutputStream(this.keyStorePath.toFile()), getKeyStorePassword());
+		} catch (Exception saveException) {
+			getLogger().severe("Couldn't store KeyStore to Storage! " + saveException.getMessage());
+		}
+	}
+	private void storeTrustStore(KeyStore truststore) {
+		try {
+			this.trustStore = truststore;
+			truststore.store(new FileOutputStream(this.trustStorePath.toFile()), getKeyStorePassword());
+		} catch (Exception saveException) {
+			getLogger().severe("Couldn't store TrustStore to Storage! " + saveException.getMessage());
+		}
+	}
+
+	private char[] getKeyStorePassword(){
+		return this.keyStorePassword.toCharArray();
+	}
+
+	@Override
+	public KeyManagerFactory getKeyManagerFactory(String alias) throws Exception {
+		KeyManagerFactory keyManagerFactory = new CustomKeyManagerFactory(alias);
+		try {
+			keyManagerFactory.init(this.keyStore, getKeyStorePassword());
+		} catch (Exception e) {
+			throw new Exception("Could not retrieve KeyManagerFactory: "+e.getMessage());
+		}
+
+		return keyManagerFactory;
+	}
+	@Override
+	public TrustManagerFactory getTrustManagerFactory() throws Exception {
+
+		CustomX509TrustManagerFactory tmf = new CustomX509TrustManagerFactory(this.trustStore, (KeyStore) null);
+		try {
+			tmf.init((KeyStore) null);
+		} catch (Exception e) {
+			throw new Exception("Could not retrieve KeyManagerFactory: "+e.getMessage());
+		}
+
+		return tmf;
+	}
+	private KeyStore createKeyStore(Path path) throws Exception {
```

Comment:
We tend to use `protected` methods to allow custom implementations to be created more easily